File: frontend/src/lib/types/product.ts
export type TProduct = {
  id: string;
  name: string;
  description: string;
  categoryId: string;
  categoryName: string;
  status: ProductStatus;
  isTrackInventory: boolean;
  isFeatured: boolean;
  price: number;
  comparePrice?: number;
  sku?: string;
  stock?: number;
  images: string[];
  primaryImageUrl: string;
  variantOptions: VariantOption[];
  variants: TProductVariant[];
  createdAt: string;
};

export type ProductStatus = "Active" | "Archived" | "Draft";

export type TProductVariant = {
  id: string;
  combination: { name: string; value: string }[];
  sku: string;
  price: number;
  available: number;
};

export type VariantForm = Omit<TProductVariant, "combination"> & {
  combination: Record<string, string>;
};

export type VariantOption = {
  name: string;
  values: string[];
};

File: backend/Features/Admin/Products/Models/Product.cs
using backend.Features.Admin.Products.Enums;
using backend.Features.Admin.Categories.Models;
using backend.Features.Admin.Stores.Models;
namespace backend.Features.Admin.Products.Models;

public class Product
{
    public Guid Id { get; set; }
    public Guid StoreId { get; set; }
    public Guid CategoryId { get; set; }
    public string Name { get; set; } = string.Empty;
    public string? SKU { get; set; }
    public string Description { get; set; } = string.Empty;
    public decimal Price { get; set; }
    public decimal? CompareAtPrice { get; set; }
    public decimal? CostPrice { get; set; }
    public int Stock { get; set; }
    public int LowStockThreshold { get; set; }
    public bool IsFeatured { get; set; }
    public bool IsTrackInventory { get; set; }
    public ProductStatus Status { get; set; }

    public ICollection<ProductImage> Images { get; set; } = new List<ProductImage>();
    public ICollection<ProductOption> Options { get; set; } = new List<ProductOption>();
    public ICollection<ProductVariant> Variants { get; set; } = new List<ProductVariant>();

    public DateTime CreatedAt { get; set; } = DateTime.UtcNow;
    public DateTime UpdatedAt { get; set; }
    public bool IsDeleted { get; set; }
    public DateTime? DeletedAt { get; set; }

    // Navigations Properties
    public Category Category { get; set; } = null!;
    public Store Store { get; set; } = null!;
}